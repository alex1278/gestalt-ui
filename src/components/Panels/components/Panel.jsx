import React, { Component } from 'react';
import PropTypes from 'prop-types';
import styled, { withTheme } from 'styled-components';
import { FontIcon } from 'react-md';
import { Title } from 'components/Typography';

const PanelWrapper = styled.div`
  width: 100%;
  box-shadow: 0 2px 2px 0 rgba(0,0,0,.14), 0 1px 5px 0 rgba(0,0,0,.12), 0 3px 1px -2px rgba(0,0,0,.2);
`;

const Header = styled.div`
  display: flex;
  align-items: center;
  box-sizing: border-box;
  height: 4em;
  background-color: ${props => props.theme.colors['$md-grey-50']};
  border-top: 1px solid ${props => props.theme.colors['$md-grey-200']};
  border-bottom: 1px solid ${props => props.theme.colors['$md-grey-200']};
  font-weight: 700;

  &:hover {
    cursor: pointer;
  }
`;

const Content = styled.div`
  background-color: ${props => props.theme.colors['$md-white']};
  ${props => (props.noPadding ? 'padding: 0' : 'padding: 1em')};
  border-bottom: 1px solid ${props => props.theme.colors['$md-grey-200']};
  width: 100%;
  display: ${props => (props.expanded ? 'block' : 'none')};
  ${props => (props.minHeight && `min-height: ${props.minHeight}`)};
`;

const ExpanderIcon = styled(FontIcon)`
  padding: .5em;
  transition: transform 100ms ease;
  transform: ${props => (props.expanded ? 'rotate(0)' : 'rotate(-90deg)')};
`;

class Panel extends Component {
  static propTypes = {
    title: PropTypes.string.isRequired,
    children: PropTypes.any.isRequired,
    defaultExpanded: PropTypes.bool,
    noPadding: PropTypes.bool,
    minHeight: PropTypes.oneOf([PropTypes.bool, PropTypes.string]),
  };

  static defaultProps = {
    defaultExpanded: true,
    noPadding: false,
    minHeight: false,
  };

  state = { isExpanded: this.props.defaultExpanded };

  toggle = () => this.setState({ isExpanded: !this.state.isExpanded });

  render() {
    return (
      <PanelWrapper>
        <Header onClick={this.toggle}>
          <ExpanderIcon expanded={this.state.isExpanded}>expand_more</ExpanderIcon>
          {this.props.title && <Title>{this.props.title}</Title>}
        </Header>
        <Content
          expanded={this.state.isExpanded}
          noPadding={this.props.noPadding}
          minHeight={this.props.minHeight}
        >
          {this.props.children}
        </Content>
      </PanelWrapper>
    );
  }
}

export default withTheme(Panel);
