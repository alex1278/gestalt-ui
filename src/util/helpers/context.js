/**
 * generateContextEntityState
 * @param {*} params
 */
export function generateContextEntityState(params) {
  const entity = {
    id: params.environmentId || params.workspaceId || null,
    key: null,
  };

  if (params.workspaceId && params.environmentId) {
    entity.key = 'environments';
  }

  if (params.workspaceId && !params.environmentId) {
    entity.key = 'workspaces';
  }

  // if (params.workspaceId && params.environmentId && params.providerId) {
  //   entity.id = params.providerId;
  //   entity.key = 'providers';
  // }

  return entity;
}
