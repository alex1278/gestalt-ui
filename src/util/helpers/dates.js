import moment from 'moment';

export const getNow = () => moment();

export const getAgo = (num = 0, unit = 'day') =>
  moment().subtract(num, unit);

export const isRecentWithin = (timestamp, num, unit) =>
  moment(timestamp).isBetween(getAgo(num, unit), getNow());
