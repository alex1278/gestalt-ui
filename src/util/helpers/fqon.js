// sad hack for parsing url

export default function fqonParent(organization) {
  if (organization.org) {
    const urlParser = document.createElement('a');
    urlParser.href = organization.org.href;
    return urlParser.pathname.replace('/', '').split('/')[0];
  }

  return organization;
}
