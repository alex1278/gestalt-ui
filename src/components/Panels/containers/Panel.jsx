import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { Card } from 'components/Cards';
import styled, { css } from 'styled-components';
import { DotActivity } from 'components/ProgressIndicators';
import Header from '../components/Header';
import Content from '../components/Content';

const PanelWrapper = styled(({ fill, error, expanded, noShadow, noBackground, ...rest }) => <Card noShadow={noShadow} {...rest} />)`
  position: relative;
  width: 100%;
  ${props => props.fill && props.expanded && 'height: 100%'};
  ${props => props.error && !props.expanded && css`
    box-shadow: 0 0 2px 2px ${props.theme.colors['$md-red-500']};
  `};
`;

class Panel extends PureComponent {
  static propTypes = {
    title: PropTypes.string,
    children: PropTypes.oneOfType([
      PropTypes.arrayOf(PropTypes.node),
      PropTypes.node
    ]),
    defaultExpanded: PropTypes.bool,
    noPadding: PropTypes.bool,
    minHeight: PropTypes.oneOfType([PropTypes.bool, PropTypes.string]),
    maxHeight: PropTypes.oneOfType([PropTypes.bool, PropTypes.string]),
    pending: PropTypes.bool,
    expandable: PropTypes.bool,
    count: PropTypes.number,
    error: PropTypes.bool,
    noShadow: PropTypes.bool,
    icon: PropTypes.oneOfType([PropTypes.func, PropTypes.node, PropTypes.bool]),
    fill: PropTypes.bool,
    disabled: PropTypes.bool,
  };

  static defaultProps = {
    title: null,
    defaultExpanded: true,
    noPadding: false,
    minHeight: false,
    maxHeight: false,
    children: null,
    pending: false,
    expandable: true,
    count: 0,
    error: false,
    noShadow: false,
    icon: false,
    fill: false,
    disabled: false,
  };

  state = { isExpanded: this.props.defaultExpanded };

  toggle = () => this.setState(prevState => ({ isExpanded: !prevState.isExpanded }));

  render() {
    const { title, noPadding, minHeight, maxHeight, pending, children, expandable, count, error, noShadow, icon, fill, disabled } = this.props;
    const { isExpanded } = this.state;

    return (
      <PanelWrapper error={error} noShadow={noShadow} fill={fill} expanded={isExpanded}>
        {!!title &&
        <Header
          title={title}
          onClick={expandable ? this.toggle : null}
          expandable={expandable}
          expanded={isExpanded}
          count={count}
          icon={icon}
          noShadow={noShadow}
        />}
        <Content
          isExpanded={isExpanded}
          noPadding={noPadding}
          minHeight={minHeight}
          maxHeight={maxHeight}
          disabled={disabled}
        >
          {pending ? <DotActivity centered size={2} /> : children}
        </Content>
      </PanelWrapper>
    );
  }
}

export default Panel;
