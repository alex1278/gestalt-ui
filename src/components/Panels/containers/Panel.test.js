import React from 'react';
import { fireEvent } from 'react-testing-library';
import Panel from './Panel';
import { renderWithTheme } from '../../../../test/helpers';

describe('(Panels) Panel', () => {
  it('renders component without exploding', () => {
    const { container } = renderWithTheme(
      <Panel title="gazobazorb" />
    );

    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders component with basic props', () => {
    const { container } = renderWithTheme(
      <Panel title="gazobazorb" />
    );

    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders component with minHeight prop', () => {
    const { container } = renderWithTheme(
      <Panel title="gazobazorb" minHeight="10em" />
    );

    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders component with noPadding prop', () => {
    const { container } = renderWithTheme(
      <Panel title="gazobazorb" noPadding />
    );

    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders component with children', () => {
    const { container } = renderWithTheme(
      <Panel title="gazobazorb"><span>this is a test</span></Panel>
    );

    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders component activity status when pending', () => {
    const { container } = renderWithTheme(
      <Panel title="gazobazorb" pending><span>this is a test</span></Panel>
    );

    expect(container.firstChild).toMatchSnapshot();
  });

  it('toggles isExpanded on correct elements when the Header is clicked', () => {
    const { container, getByText } = renderWithTheme(
      <Panel title="gazobazorb"><span>this is a test</span></Panel>
    );
    const header = getByText('gazobazorb');

    fireEvent.click(header);

    expect(container.firstChild).toMatchSnapshot();
  });
});
