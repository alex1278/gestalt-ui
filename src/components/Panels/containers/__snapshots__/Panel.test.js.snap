// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Panels) Panel renders component activity status when pending 1`] = `
.c1 {
  position: relative;
  background: white;
  border-radius: 3px;
  box-shadow: 0 1px 1px 0 rgba(0,0,0,.1),0 1px 1px 0 rgba(0,0,0,.1),0 0 1px -4px rgba(0,0,0,.2);
}

.c9 {
  position: relative;
  width: 100%;
  height: 100%;
}

.c10 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  height: 100%;
}

.c10.spinner > div {
  width: 2em;
  height: 2em;
  background-color: #9e9e9e;
  border-radius: 100%;
  display: inline-block;
  -webkit-animation: bJpfxq 1.4s infinite ease-in-out both;
  animation: bJpfxq 1.4s infinite ease-in-out both;
}

.c10.spinner .bounce1 {
  -webkit-animation-delay: -0.32s;
  animation-delay: -0.32s;
}

.c10.spinner .bounce2 {
  -webkit-animation-delay: -0.16s;
  animation-delay: -0.16s;
}

.c3 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 0 1 auto;
  -ms-flex: 0 1 auto;
  flex: 0 1 auto;
  width: 100%;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  height: 100%;
}

.c5 {
  box-sizing: border-box;
  margin: 0px;
  max-height: 100%;
  -webkit-order: 0;
  -ms-flex-order: 0;
  order: 0;
  -webkit-align-self: auto;
  -ms-flex-item-align: auto;
  align-self: auto;
  --width: calc(100% - 0px);
  -webkit-flex: 0 1 var(--width);
  -ms-flex: 0 1 var(--width);
  flex: 0 1 var(--width);
  min-width: var(--width);
}

.c6 {
  margin-left: 3px;
  margin-right: 3px;
  -webkit-transition: -webkit-transform 100ms ease;
  -webkit-transition: transform 100ms ease;
  transition: transform 100ms ease;
  -webkit-transform: rotate(90deg);
  -ms-transform: rotate(90deg);
  transform: rotate(90deg);
}

.c4 {
  padding-left: 8px;
  padding-right: 8px;
  padding-top: 12px;
  padding-bottom: 12px;
  height: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c7 {
  display: inline-block;
  font-size: 16px;
  color: #424242;
}

.c2 {
  height: 48px;
  font-weight: bold;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c2:hover {
  cursor: pointer;
}

.c8 {
  width: 100%;
  display: block;
  padding: 1em;
}

.c0 {
  position: relative;
  width: 100%;
}

<div
  class="c0 c1"
>
  <header
    class="c2"
  >
    <div
      class="c3"
    >
      <div
        class="c4 c5"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root-1 MuiSvgIcon-colorAction-4 c6"
          focusable="false"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
          />
          <path
            d="M0 0h24v24H0z"
            fill="none"
          />
        </svg>
        <div
          class="c7"
        >
          gazobazorb
        </div>
      </div>
    </div>
  </header>
  <div
    class="c8"
  >
    <div
      class="c9"
    >
      <div
        class="spinner c10"
        size="2"
      >
        <div
          class="bounce1"
        />
        <div
          class="bounce2"
        />
        <div
          class="bounce3"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`(Panels) Panel renders component with basic props 1`] = `
.c1 {
  position: relative;
  background: white;
  border-radius: 3px;
  box-shadow: 0 1px 1px 0 rgba(0,0,0,.1),0 1px 1px 0 rgba(0,0,0,.1),0 0 1px -4px rgba(0,0,0,.2);
}

.c3 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 0 1 auto;
  -ms-flex: 0 1 auto;
  flex: 0 1 auto;
  width: 100%;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  height: 100%;
}

.c5 {
  box-sizing: border-box;
  margin: 0px;
  max-height: 100%;
  -webkit-order: 0;
  -ms-flex-order: 0;
  order: 0;
  -webkit-align-self: auto;
  -ms-flex-item-align: auto;
  align-self: auto;
  --width: calc(100% - 0px);
  -webkit-flex: 0 1 var(--width);
  -ms-flex: 0 1 var(--width);
  flex: 0 1 var(--width);
  min-width: var(--width);
}

.c6 {
  margin-left: 3px;
  margin-right: 3px;
  -webkit-transition: -webkit-transform 100ms ease;
  -webkit-transition: transform 100ms ease;
  transition: transform 100ms ease;
  -webkit-transform: rotate(90deg);
  -ms-transform: rotate(90deg);
  transform: rotate(90deg);
}

.c4 {
  padding-left: 8px;
  padding-right: 8px;
  padding-top: 12px;
  padding-bottom: 12px;
  height: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c7 {
  display: inline-block;
  font-size: 16px;
  color: #424242;
}

.c2 {
  height: 48px;
  font-weight: bold;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c2:hover {
  cursor: pointer;
}

.c8 {
  width: 100%;
  display: block;
  padding: 1em;
}

.c0 {
  position: relative;
  width: 100%;
}

<div
  class="c0 c1"
>
  <header
    class="c2"
  >
    <div
      class="c3"
    >
      <div
        class="c4 c5"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root-1 MuiSvgIcon-colorAction-4 c6"
          focusable="false"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
          />
          <path
            d="M0 0h24v24H0z"
            fill="none"
          />
        </svg>
        <div
          class="c7"
        >
          gazobazorb
        </div>
      </div>
    </div>
  </header>
  <div
    class="c8"
  />
</div>
`;

exports[`(Panels) Panel renders component with children 1`] = `
.c1 {
  position: relative;
  background: white;
  border-radius: 3px;
  box-shadow: 0 1px 1px 0 rgba(0,0,0,.1),0 1px 1px 0 rgba(0,0,0,.1),0 0 1px -4px rgba(0,0,0,.2);
}

.c3 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 0 1 auto;
  -ms-flex: 0 1 auto;
  flex: 0 1 auto;
  width: 100%;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  height: 100%;
}

.c5 {
  box-sizing: border-box;
  margin: 0px;
  max-height: 100%;
  -webkit-order: 0;
  -ms-flex-order: 0;
  order: 0;
  -webkit-align-self: auto;
  -ms-flex-item-align: auto;
  align-self: auto;
  --width: calc(100% - 0px);
  -webkit-flex: 0 1 var(--width);
  -ms-flex: 0 1 var(--width);
  flex: 0 1 var(--width);
  min-width: var(--width);
}

.c6 {
  margin-left: 3px;
  margin-right: 3px;
  -webkit-transition: -webkit-transform 100ms ease;
  -webkit-transition: transform 100ms ease;
  transition: transform 100ms ease;
  -webkit-transform: rotate(90deg);
  -ms-transform: rotate(90deg);
  transform: rotate(90deg);
}

.c4 {
  padding-left: 8px;
  padding-right: 8px;
  padding-top: 12px;
  padding-bottom: 12px;
  height: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c7 {
  display: inline-block;
  font-size: 16px;
  color: #424242;
}

.c2 {
  height: 48px;
  font-weight: bold;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c2:hover {
  cursor: pointer;
}

.c8 {
  width: 100%;
  display: block;
  padding: 1em;
}

.c0 {
  position: relative;
  width: 100%;
}

<div
  class="c0 c1"
>
  <header
    class="c2"
  >
    <div
      class="c3"
    >
      <div
        class="c4 c5"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root-1 MuiSvgIcon-colorAction-4 c6"
          focusable="false"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
          />
          <path
            d="M0 0h24v24H0z"
            fill="none"
          />
        </svg>
        <div
          class="c7"
        >
          gazobazorb
        </div>
      </div>
    </div>
  </header>
  <div
    class="c8"
  >
    <span>
      this is a test
    </span>
  </div>
</div>
`;

exports[`(Panels) Panel renders component with minHeight prop 1`] = `
.c1 {
  position: relative;
  background: white;
  border-radius: 3px;
  box-shadow: 0 1px 1px 0 rgba(0,0,0,.1),0 1px 1px 0 rgba(0,0,0,.1),0 0 1px -4px rgba(0,0,0,.2);
}

.c3 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 0 1 auto;
  -ms-flex: 0 1 auto;
  flex: 0 1 auto;
  width: 100%;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  height: 100%;
}

.c5 {
  box-sizing: border-box;
  margin: 0px;
  max-height: 100%;
  -webkit-order: 0;
  -ms-flex-order: 0;
  order: 0;
  -webkit-align-self: auto;
  -ms-flex-item-align: auto;
  align-self: auto;
  --width: calc(100% - 0px);
  -webkit-flex: 0 1 var(--width);
  -ms-flex: 0 1 var(--width);
  flex: 0 1 var(--width);
  min-width: var(--width);
}

.c6 {
  margin-left: 3px;
  margin-right: 3px;
  -webkit-transition: -webkit-transform 100ms ease;
  -webkit-transition: transform 100ms ease;
  transition: transform 100ms ease;
  -webkit-transform: rotate(90deg);
  -ms-transform: rotate(90deg);
  transform: rotate(90deg);
}

.c4 {
  padding-left: 8px;
  padding-right: 8px;
  padding-top: 12px;
  padding-bottom: 12px;
  height: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c7 {
  display: inline-block;
  font-size: 16px;
  color: #424242;
}

.c2 {
  height: 48px;
  font-weight: bold;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c2:hover {
  cursor: pointer;
}

.c8 {
  width: 100%;
  display: block;
  padding: 1em;
  min-height: 10em;
}

.c0 {
  position: relative;
  width: 100%;
}

<div
  class="c0 c1"
>
  <header
    class="c2"
  >
    <div
      class="c3"
    >
      <div
        class="c4 c5"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root-1 MuiSvgIcon-colorAction-4 c6"
          focusable="false"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
          />
          <path
            d="M0 0h24v24H0z"
            fill="none"
          />
        </svg>
        <div
          class="c7"
        >
          gazobazorb
        </div>
      </div>
    </div>
  </header>
  <div
    class="c8"
  />
</div>
`;

exports[`(Panels) Panel renders component with noPadding prop 1`] = `
.c1 {
  position: relative;
  background: white;
  border-radius: 3px;
  box-shadow: 0 1px 1px 0 rgba(0,0,0,.1),0 1px 1px 0 rgba(0,0,0,.1),0 0 1px -4px rgba(0,0,0,.2);
}

.c3 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 0 1 auto;
  -ms-flex: 0 1 auto;
  flex: 0 1 auto;
  width: 100%;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  height: 100%;
}

.c5 {
  box-sizing: border-box;
  margin: 0px;
  max-height: 100%;
  -webkit-order: 0;
  -ms-flex-order: 0;
  order: 0;
  -webkit-align-self: auto;
  -ms-flex-item-align: auto;
  align-self: auto;
  --width: calc(100% - 0px);
  -webkit-flex: 0 1 var(--width);
  -ms-flex: 0 1 var(--width);
  flex: 0 1 var(--width);
  min-width: var(--width);
}

.c6 {
  margin-left: 3px;
  margin-right: 3px;
  -webkit-transition: -webkit-transform 100ms ease;
  -webkit-transition: transform 100ms ease;
  transition: transform 100ms ease;
  -webkit-transform: rotate(90deg);
  -ms-transform: rotate(90deg);
  transform: rotate(90deg);
}

.c4 {
  padding-left: 8px;
  padding-right: 8px;
  padding-top: 12px;
  padding-bottom: 12px;
  height: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c7 {
  display: inline-block;
  font-size: 16px;
  color: #424242;
}

.c2 {
  height: 48px;
  font-weight: bold;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c2:hover {
  cursor: pointer;
}

.c8 {
  width: 100%;
  display: block;
  padding: 0;
}

.c0 {
  position: relative;
  width: 100%;
}

<div
  class="c0 c1"
>
  <header
    class="c2"
  >
    <div
      class="c3"
    >
      <div
        class="c4 c5"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root-1 MuiSvgIcon-colorAction-4 c6"
          focusable="false"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
          />
          <path
            d="M0 0h24v24H0z"
            fill="none"
          />
        </svg>
        <div
          class="c7"
        >
          gazobazorb
        </div>
      </div>
    </div>
  </header>
  <div
    class="c8"
  />
</div>
`;

exports[`(Panels) Panel renders component without exploding 1`] = `
.c1 {
  position: relative;
  background: white;
  border-radius: 3px;
  box-shadow: 0 1px 1px 0 rgba(0,0,0,.1),0 1px 1px 0 rgba(0,0,0,.1),0 0 1px -4px rgba(0,0,0,.2);
}

.c3 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 0 1 auto;
  -ms-flex: 0 1 auto;
  flex: 0 1 auto;
  width: 100%;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  height: 100%;
}

.c5 {
  box-sizing: border-box;
  margin: 0px;
  max-height: 100%;
  -webkit-order: 0;
  -ms-flex-order: 0;
  order: 0;
  -webkit-align-self: auto;
  -ms-flex-item-align: auto;
  align-self: auto;
  --width: calc(100% - 0px);
  -webkit-flex: 0 1 var(--width);
  -ms-flex: 0 1 var(--width);
  flex: 0 1 var(--width);
  min-width: var(--width);
}

.c6 {
  margin-left: 3px;
  margin-right: 3px;
  -webkit-transition: -webkit-transform 100ms ease;
  -webkit-transition: transform 100ms ease;
  transition: transform 100ms ease;
  -webkit-transform: rotate(90deg);
  -ms-transform: rotate(90deg);
  transform: rotate(90deg);
}

.c4 {
  padding-left: 8px;
  padding-right: 8px;
  padding-top: 12px;
  padding-bottom: 12px;
  height: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c7 {
  display: inline-block;
  font-size: 16px;
  color: #424242;
}

.c2 {
  height: 48px;
  font-weight: bold;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c2:hover {
  cursor: pointer;
}

.c8 {
  width: 100%;
  display: block;
  padding: 1em;
}

.c0 {
  position: relative;
  width: 100%;
}

<div
  class="c0 c1"
>
  <header
    class="c2"
  >
    <div
      class="c3"
    >
      <div
        class="c4 c5"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root-1 MuiSvgIcon-colorAction-4 c6"
          focusable="false"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
          />
          <path
            d="M0 0h24v24H0z"
            fill="none"
          />
        </svg>
        <div
          class="c7"
        >
          gazobazorb
        </div>
      </div>
    </div>
  </header>
  <div
    class="c8"
  />
</div>
`;

exports[`(Panels) Panel toggles isExpanded on correct elements when the Header is clicked 1`] = `
.c1 {
  position: relative;
  background: white;
  border-radius: 3px;
  box-shadow: 0 1px 1px 0 rgba(0,0,0,.1),0 1px 1px 0 rgba(0,0,0,.1),0 0 1px -4px rgba(0,0,0,.2);
}

.c3 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 0 1 auto;
  -ms-flex: 0 1 auto;
  flex: 0 1 auto;
  width: 100%;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  height: 100%;
}

.c5 {
  box-sizing: border-box;
  margin: 0px;
  max-height: 100%;
  -webkit-order: 0;
  -ms-flex-order: 0;
  order: 0;
  -webkit-align-self: auto;
  -ms-flex-item-align: auto;
  align-self: auto;
  --width: calc(100% - 0px);
  -webkit-flex: 0 1 var(--width);
  -ms-flex: 0 1 var(--width);
  flex: 0 1 var(--width);
  min-width: var(--width);
}

.c6 {
  margin-left: 3px;
  margin-right: 3px;
  -webkit-transition: -webkit-transform 100ms ease;
  -webkit-transition: transform 100ms ease;
  transition: transform 100ms ease;
  -webkit-transform: rotate(90deg);
  -ms-transform: rotate(90deg);
  transform: rotate(90deg);
}

.c4 {
  padding-left: 8px;
  padding-right: 8px;
  padding-top: 12px;
  padding-bottom: 12px;
  height: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c7 {
  display: inline-block;
  font-size: 16px;
  color: #424242;
}

.c2 {
  height: 48px;
  font-weight: bold;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c2:hover {
  cursor: pointer;
}

.c8 {
  width: 100%;
  display: block;
  padding: 1em;
}

.c0 {
  position: relative;
  width: 100%;
}

<div
  class="c0 c1"
>
  <header
    class="c2"
  >
    <div
      class="c3"
    >
      <div
        class="c4 c5"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root-1 MuiSvgIcon-colorAction-4 c6"
          focusable="false"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
          />
          <path
            d="M0 0h24v24H0z"
            fill="none"
          />
        </svg>
        <div
          class="c7"
        >
          gazobazorb
        </div>
      </div>
    </div>
  </header>
  <div
    class="c8"
  >
    <span>
      this is a test
    </span>
  </div>
</div>
`;
